tests/shared/src/test/scala-2.13/scala/meta/tests/parsers/dotty/Scala3PositionSuite.scala

```diff
@@ -301,13 +301,58 @@ class Scala3PositionSuite extends BasePositionSuite(dialects.Scala3) {
        |    42
        |  def b: String =
        |    "b"
-       |Self def
-       |Name.Anonymous def
+       |Self   @@def a: Int =
        |Defn.Def def a: Int =
        |    42
        |Defn.Def def b: String =
        |    "b"
        |Lit.String "b"
        |""".stripMargin
   )
+
+  checkPositions[Stat](
+    """|object a:
+       |   def foo =
+       |     try foo
+       |     catch
+       |       case a =>
+       |       case b =>
+       |         st1
+       |         st2
+       |     finally bar
+       |""".stripMargin,
+    """|Template :
+       |   def foo =
+       |     try foo
+       |     catch
+       |       case a =>
+       |       case b =>
+       |         st1
+       |         st2
+       |     finally bar
+       |Self    @@def foo =
```

Comment:
why is self pointing to `def`?

Reply:
It's just an empty Position, it needs to be somewhere so it's at the start of the template, it contains no tokens. It's the same as in the other tests.